modules/pointcloudio/src/coders_obj.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef _CODERS_OBJ_H_
+#define _CODERS_OBJ_H_
+
+#include "coders_base.hpp" 
+
+namespace cv 
+{
+namespace pc
+{
+
+class ObjDecoder CV_FINAL : public BasePointCloudDecoder
+{
+public:
+
+    ObjDecoder();
+    ~ObjDecoder() CV_OVERRIDE;
+
+    void readData( std::vector<Point3f>& points, std::vector<Point3f>& normals ) CV_OVERRIDE;
+
+    PointCloudDecoder newDecoder() const CV_OVERRIDE;
```

Comment:
It looks like `newDecoder` method is redundant for both cases as it does noting than just object construction.